pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go

```diff
@@ -137,58 +140,39 @@ func (s *AttestorSuite) TestAttestFailsWhenAttestedBefore() {
 	})
 
 	token := s.signToken(s.fooSigner, "NS1", "SA1")
-	s.requireAttestError(makeAttestRequest("FOO", token),
-		"k8s-sat: SAT has already been used to attest an agent with the same UUID")
-}
-
-func (s *AttestorSuite) TestAttestFailsWithNoAttestationData() {
-	s.requireAttestError(&nodeattestorv0.AttestRequest{},
-		"k8s-sat: missing attestation data")
-}
-
-func (s *AttestorSuite) TestAttestFailsWithWrongAttestationDataType() {
-	s.requireAttestError(&nodeattestorv0.AttestRequest{
-		AttestationData: &common.AttestationData{
-			Type: "blah",
-		},
-	}, `k8s-sat: unexpected attestation data type "blah"`)
+	s.requireAttestError(makePayload("FOO", token),
+		codes.PermissionDenied,
+		"nodeattestor(k8s_sat): SAT has already been used to attest an agent with the same UUID")
 }
 
-func (s *AttestorSuite) TestAttestFailsWithNoAttestationDataPayload() {
```

Comment:
We are losing coverage for this case.

Reply:
This case isn't actually easy to test anymore, since the facade now prevents sending an empty payload.